### PR TITLE
fix the bug in CDN edge node and udpated version

### PR DESCRIPTION
**Pull request by csmengwan**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/dev/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as a link to the swagger spec, used to generate the code.
- [x] The `project.json` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@csmengwan__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (qinwa). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: src/ResourceManagement/Cdn/Microsoft.Azure.Management.Cdn/Generated/Models/CidrIpAddress.cs [hunk 1]
```diff
@@ -46,7 +46,7 @@ public CidrIpAddress(string baseIpAddress = default(string), string prefixLength
         /// Gets or sets the length of the prefix of the ip address.
         /// </summary>
         [JsonProperty(PropertyName = "prefixLength")]
-        public string PrefixLength { get; set; }
+        public int? PrefixLength { get; set; }
```
Comment (cormacpayne) [inline src/ResourceManagement/Cdn/Microsoft.Azure.Management.Cdn/Generated/Models/CidrIpAddress.cs]: @csmengwan changing the type of this property is a breaking change

File: src/ResourceManagement/Cdn/Microsoft.Azure.Management.Cdn/Generated/Models/EdgeNode.cs [hunk 2]
```diff
@@ -29,36 +32,38 @@ public partial class EdgeNode
...
-        /// <param name="resourceGroup">The resource group of the edge
-        /// node.</param>
+        /// <param name="location">Resource location.</param>
         /// <param name="ipAddressGroups">List of ip address groups.</param>
-        public EdgeNode(string name = default(string), string resourceGroup = default(string), IList<IpAddressGroup> ipAddressGroups = default(IList<IpAddressGroup>))
+        /// <param name="id">Resource ID.</param>
+        /// <param name="name">Resource name.</param>
+        /// <param name="type">Resource type.</param>
+        /// <param name="tags">Resource tags.</param>
+        public EdgeNode(string location, IList<IpAddressGroup> ipAddressGroups, string id = default(string), string name = default(string), string type = default(string), IDictionary<string, string> tags = default(IDictionary<string, string>))
```
Comment (cormacpayne) [inline src/ResourceManagement/Cdn/Microsoft.Azure.Management.Cdn/Generated/Models/EdgeNode.cs]: @csmengwan changing the order of parameters in the constructor is a breaking change

File: src/ResourceManagement/Cdn/Microsoft.Azure.Management.Cdn/Generated/Models/EdgeNode.cs [hunk 3]
```diff
@@ -29,36 +32,38 @@ public partial class EdgeNode
...
-            Name = name;
-            ResourceGroup = resourceGroup;
             IpAddressGroups = ipAddressGroups;
         }
 
         /// <summary>
-        /// Gets or sets ip adress group that contains Ipv4 and Ipv6 addresses
-        /// </summary>
-        [JsonProperty(PropertyName = "name")]
-        public string Name { get; set; }
```
Comment (cormacpayne) [inline src/ResourceManagement/Cdn/Microsoft.Azure.Management.Cdn/Generated/Models/EdgeNode.cs]: @csmengwan removing this property is a breaking change

File: src/ResourceManagement/Cdn/Microsoft.Azure.Management.Cdn/Generated/Models/EdgeNode.cs [hunk 4]
```diff
@@ -29,36 +32,38 @@ public partial class EdgeNode
...
-
-        /// <summary>
-        /// Gets or sets the resource group of the edge node.
-        /// </summary>
-        [JsonProperty(PropertyName = "resourceGroup")]
-        public string ResourceGroup { get; set; }
```
Comment (cormacpayne) [inline src/ResourceManagement/Cdn/Microsoft.Azure.Management.Cdn/Generated/Models/EdgeNode.cs]: @csmengwan removing this property is a breaking change

**Review by cormacpayne — CHANGES_REQUESTED** [review]
@csmengwan a couple of comments

Also, was this code generated based off of swagger? If so, you will need to specify what version of AutoRest and swagger spec were used in `generate.cmd`

**Review by cormacpayne — APPROVED** [review]

**Pull request merged**